src/main/java/gov/nasa/pds/tools/validate/rule/pds4/ArrayValidator.java

```diff
@@ -73,6 +74,8 @@ public boolean validateDataObjectDefinition() {
 
   @Override
   public boolean validateDataObjectContents() throws InvalidTableException, IOException {
+	if (EveryNCounter.getInstance().getValueThenIncrement() % this.context.getEveryN() != 0) return true;
```

Comment:
@al-niessner the increment needs to happen at the file level, not the data object level.  so somewhere before the loop through the data objects. maybe [here](https://github.com/NASA-PDS/validate/blob/issue_1/src/main/java/gov/nasa/pds/tools/validate/rule/pds4/DataDefinitionAndContentValidationRule.java#L54)?

Reply:
or maybe after the loop [here](https://github.com/NASA-PDS/validate/blob/issue_1/src/main/java/gov/nasa/pds/tools/validate/rule/pds4/DataDefinitionAndContentValidationRule.java#L125) depending up how you are calculating everyN

Reply:
@jordanpadams 

It does, essentially. The object is global so every DataSomethingSomething that process a file to either table or array increments the same global counter. The code is not written to do a table and an array in a single file, making the increment per file. Just made more sense to do the increment locally rather than far away then test else where.

Reply:
@jordanpadams 

Oh, each object is checked but there can be more than one object per file in DataSomethingSomething.validate(). Yeah, I can change it.

Reply:
@jordanpadams fixed